uaclient/contract.py

```diff
@@ -492,3 +531,48 @@ def get_contract_information(cfg: UAConfig, token: str) -> Dict[str, Any]:
     """Query contract information for a specific token"""
     client = UAContractClient(cfg)
     return client.request_contract_information(token)
+
+
+def is_contract_changed(cfg: UAConfig) -> bool:
+    orig_token = cfg.machine_token
+    orig_entitlements = cfg.entitlements
+    machine_token = orig_token.get("machineToken", "")
+    contract_id = (
+        orig_token.get("machineTokenInfo", {})
+        .get("contractInfo", {})
+        .get("id", "")
+    )
+    contract_client = UAContractClient(cfg)
+    resp = contract_client.get_updated_contract_info(
+        machine_token, contract_id
+    )
+    try:
+        resp_expiry = (
+            resp.get("machineTokenInfo", {})
+            .get("contractInfo", {})
+            .get("effectiveTo", "0001-01-01T00:00:00+00:00")
+        )
+        resp["machineTokenInfo"]["contractInfo"][
+            "effectiveTo"
+        ] = util.parse_rfc3339_date(resp_expiry)
+    except KeyError as e:
+        logging.error(str(e))
+        return False
+    except ValueError as e:
+        logging.error(str(e))
+        return False
+    new_expiry = (
+        resp.get("machineTokenInfo", {})
+        .get("contractInfo", {})
+        .get("effectiveTo", cfg.contract_expiry_datetime)
+    )
```

Comment:
Can this be simplified to
```suggestion
    resp_expiry = (
        resp.get("machineTokenInfo", {})
        .get("contractInfo", {})
        .get("effectiveTo", None)
    )
    new_expiry = util.parse_rfc3339_date(resp_expiry) if resp_expiry else cfg.contract_expiry_datetime
```